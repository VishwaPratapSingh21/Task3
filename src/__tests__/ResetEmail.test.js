import React from 'react';
import { shallow, enzyme } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import PasswordResetEmail from '../components/auth/ResetEmail';

describe('ResetEmail component', () => {
  const wrapper = shallow(<PasswordResetEmail />);
  const preventDefault = jest.fn();

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('renders modal in divs', () => {
    expect(wrapper.find('div')).toHaveLength(6);
  });
  it('renders state initially', () => {
    expect(wrapper.state().email).toEqual('');
  });
  it('renders a form', () => {
    expect(wrapper.find('form')).toHaveLength(1);
    expect(wrapper.find('form').simulate('submit', {preventDefault}))
    expect(preventDefault).toBeCalled();
    expect(wrapper.find('[name="email"]').simulate('change', {target:{name:'email', value:'had@hmail.com'}}))
  });

  it('has form fields', () => {
    expect(wrapper.find('[name="email"]')).toHaveLength(1);
    expect(wrapper.find('#sendemail')).toHaveLength(1);
  });

  it('Form email field update when state changes', () => {
    wrapper.setState({ email: 'mail' });
    expect(wrapper.find('[name="email"]').props().value).toEqual('mail');
  });
});
