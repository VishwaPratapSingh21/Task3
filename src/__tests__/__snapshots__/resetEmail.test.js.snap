// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetEmail component renders properly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalLabel"
  className="modal fade"
  id="forgot_password"
  role="dialog"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
        >
          Reset password
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          id="close-email-modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="modal-body"
        >
          <input
            aria-describedby="email"
            className="form-control"
            name="email"
            onChange={[Function]}
            placeholder="example@gmail.com"
            type="email"
            value=""
          />
        </div>
        <div
          className="modal-footer"
        >
          <button
            className="btn btn-primary"
            id="sendemail"
            type="submit"
          >
            Send Email
          </button>
          <button
            className="btn btn-secondary"
            data-dismiss="modal"
            type="button"
          >
            Cancel
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
